code/game/objects/items/melee/transforming.dm

```diff
@@ -99,3 +103,59 @@
 	if(force_on + increment > max)
 		return COMPONENT_BLOCK_SHARPEN_MAXED
 	sharpened_bonus = increment
+
+/obj/item/melee/transforming/butter_fly
+	name = "butterfly knife"
+	icon_state = "butterfly"
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	desc = "funny desc here."
+	flags_1 = CONDUCT_1
+	force = 3
+	w_class = WEIGHT_CLASS_SMALL
+	throwforce = 5
+	throw_speed = 3
+	throw_range = 6
+	bare_wound_bonus = 5
+	stealthy_audio = FALSE
+	force_on = 15
+	icon_state_on = "butterfly_ext"
+	hitsound_on = 'sound/weapons/bladeslice.ogg'
+	attack_verb_on = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "cuts")
+	attack_verb_off = list("pokes")
+	w_class_on = WEIGHT_CLASS_SMALL
+	sound_when_on = 'sound/weapons/batonextend.ogg'
+	sound_when_off = 'sound/weapons/batonextend.ogg'
+	///damage dealt on backstab
+	var/backstab_damage = 40
+	COOLDOWN_DECLARE(stab_cooldown)
+	///cooldown on stabbing
+	var/stab_cooldown_time = 3 SECONDS
+
+/obj/item/melee/transforming/butter_fly/attack_alt(mob/living/victim, mob/living/user, params)
```

Comment:
why make this an alt attack instead of just the main attack (with a butterknifing being done if the facings aren't lined up)

Reply:
its in as a risk-reward thing, to make backstab used only when ur likely to succeed instead of left clicking  spam

Reply:
maybe make the cooldown only apply if you fail your backstab, then? that way, you could still do cool chainstabs